netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -77,6 +85,100 @@ _Guarded_by_(_net_ebpf_ext_sock_addr_lock) static LIST_ENTRY _net_ebpf_ext_redir
 _Guarded_by_(_net_ebpf_ext_sock_addr_lock) static LIST_ENTRY _net_ebpf_ext_connect_context_list;
 static uint32_t _net_ebpf_ext_connect_context_count = 0;
 
+static SECURITY_DESCRIPTOR* _net_ebpf_ext_security_descriptor_admin = NULL;
+static ACL* _net_ebpf_ext_dacl_admin = NULL;
+
+//
+// sock_addr helper functions.
+//
+static uint32_t
+_get_process_id()
+{
+    return (uint32_t)(uintptr_t)PsGetCurrentProcessId();
+}
+
+static uint32_t
+_get_thread_id()
+{
+    return (uint32_t)(uintptr_t)PsGetCurrentThreadId();
+}
+
+static uint64_t
+_ebpf_sock_addr_get_current_pid_tgid()
+{
+    net_ebpf_sock_addr_t* sock_addr_ctx = NULL;
+    ASSERT(_net_ebpf_sock_addr_get_program_context != NULL);
+    _net_ebpf_sock_addr_get_program_context((void**)&sock_addr_ctx);
+    ASSERT(sock_addr_ctx != NULL);
```

Comment:
Not needed IMO. In the case `sock_addr_ctx == NULL` ; you may want to trace something.

Reply:
I can add trace, but I want to keep the ASSERT. This will help catch any regression in this API.